rest/config_startup.go

```diff
@@ -128,8 +145,10 @@ type APIConfig struct {
 	CompressResponses  *bool `json:"compress_responses,omitempty"   help:"If false, disables compression of HTTP responses"`
 	HideProductVersion *bool `json:"hide_product_version,omitempty" help:"Whether product versions removed from Server headers and REST API responses"`
 
-	HTTPS HTTPSConfig      `json:"https,omitempty"`
-	CORS  *auth.CORSConfig `json:"cors,omitempty"`
+	HTTPS                          HTTPSConfig      `json:"https,omitempty"`
+	CORS                           *auth.CORSConfig `json:"cors,omitempty"`
+	HeapProfileCollectionThreshold *uint64          `json:"heap_profile_collection_threshold,omitempty" help:"Threshold in bytes for collecting heap profiles automatically. If set, Sync Gateway will collect a memory profile when it exceeds this value. The default value will be set to 85% of the lesser of cgroup or system memory."`
```

Comment:
I'm not sure these belong in APIConfig (which has to do with the REST APIs).  Was there a reason to put it here, as opposed to (say) BootstrapConfig?

Reply:
I think it doesn't belong in the bootstrap config because it doesn't have to do with couchbase server. I moved it to the toplevel.

I don't love that either, but I agree it doesn't below in API.